### PR TITLE
Permissioning + DVD transfers

### PR DESCRIPTION
**Pull request by Joachim-Lebrun**
- modification of permissioning smart contracts 
- DVD transfers contract

### CODE REVIEW

File: contracts/roles/OwnerManager.sol [hunk 1]
```diff
@@ -219,12 +262,12 @@ contract OwnerManager is OwnerRoles {
...
      *  See {IClaimTopicsRegistry-addClaimTopic}.
      *  Requires that `_onchainID` is set as ClaimRegistryManager on the OwnerManager contract
-     *  Requires that msg.sender is a MANAGEMENT KEY on `_onchainID`
+     *  Requires that msg.sender is an ACTION KEY on `_onchainID`
      *  @param _onchainID the _onchainID contract of the caller, e.g. "i call this function and i am Bob"
      */
     function callAddClaimTopic(uint256 _claimTopic, IIdentity _onchainID) external {
         require(
-            isClaimRegistryManager(address(_onchainID)) && _onchainID.keyHasPurpose(keccak256(abi.encode(msg.sender)), 1),
+            isClaimRegistryManager(address(_onchainID)) && _onchainID.keyHasPurpose(keccak256(abi.encode(msg.sender)), 2),
```
Comment (chichke) [inline contracts/roles/OwnerManager.sol:270]: Why putting 2 instead of true or false?

Comment (Joachim-Lebrun) [reply contracts/roles/OwnerManager.sol:270]: i don't understand your comment, the `2` is the key type. 
Key type 1 is MANAGEMENT key 
key type 2 is ACTION key
key type 3 is CLAIM signing key
etc.

Comment (chichke) [reply contracts/roles/OwnerManager.sol:270]: function isClaimRegistryManager(address _owner) public view returns (bool)
Returns a boolean

Comment (chichke) [reply contracts/roles/OwnerManager.sol:270]: I might have misunderstood something

Comment (chichke) [reply contracts/roles/OwnerManager.sol:270]: https://github.com/TokenySolutions/T-REX/blob/04d1093a41f1e208d6f6c430f4538e0117612c9a/test/ownerManager.test.js#L172

That's how it is used in the tests

Comment (chichke) [reply contracts/roles/OwnerManager.sol:270]: https://github.com/onchain-id/solidity/blob/master/contracts/interface/IERC734.sol#L103

 keyHasPurpose returns a bool as well

Comment (chichke) [reply contracts/roles/OwnerManager.sol:270]: Oh, my bad it's all good! ✌️ I just saw the parenthesis 😅

**Review by chichke — APPROVED** [review]
Good ✌️

**Comment by Mesta** [comment]
There are some conflicts, you should rebase that branch 🙄

**Comment by Joachim-Lebrun** [comment]
> There are some conflicts, you should rebase that branch 🙄

i will don't worry 🙈

File: .gitignore [hunk 2]
```diff
@@ -10,6 +10,5 @@ bin
 .vscode/
 
 # NPM
-package-lock.json
```
[the remark this answers is not in the document]
Comment (Joachim-Lebrun) [reply .gitignore:13]: we use npm, you are right package lock shouldn't be in the gitignore, on my local version it is not ignored tho... weird that it is here, i will modify thanks for reporting

**Review by Nakasar — APPROVED** [review]

**Review by Nakasar — APPROVED** [review]

**Pull request merged**